lib/yotpo/core/requests.rb

```diff
@@ -0,0 +1,64 @@
+module Yotpo
+  module Requests
+    class Endpoint
+      attr_reader :uri, :request_method, :options
+
+      def initialize(uri, **options)
+        @uri = uri
+        @options = options
+        @request_method = options.delete(:method) || :get
+      end
+
+      def request(client, params, **headers)
+        Request.new(self, params, headers).send_with(client)
+      end
+
+      def public?
+        options.key?(:public) ? !!options[:public] : false
+      end
+    end
+
+    class Request
+      attr_reader :endpoint, :uri, :params, :headers
+
+      def initialize(endpoint, params, headers)
+        @endpoint = endpoint
+        @params = Hashie::Mash.new(params)
+        @headers = headers.dup
+        @headers['X-Yotpo-Token'] = @params.delete(:utoken) || Yotpo.utoken unless endpoint.public?
+        @app_key = @params.delete(:app_key) || Yotpo.app_key
```

Comment:
Not sure how likely this scenario is but the following could happen: 

```
{ foo: '' }.delete(:foo) || 'bar'  => ""
```

Evidently, the way to fix this is by using `presence` which is not available so I will leave this call up 2 u. If you think this scenario is unlikely then you can ignore this comment.

Reply:
I think this is fine; if a bad value such as `''` is being explicitly provided, then that is what should be used and it should be resolved by the caller